packages/rum-core/src/boot/rumPublicApi.ts

```diff
@@ -201,40 +217,45 @@ export function makeRumPublicApi(
     }),
 
     setUser: monitor((newUser: User) => {
-      const sanitizedUser = sanitizeUser(newUser)
-      if (sanitizedUser) {
-        user = sanitizedUser
-      } else {
+      if (typeof newUser !== 'object' || !newUser) {
         display.error('Unsupported user:', newUser)
+      } else {
+        userContextManager.setContext(sanitizeUser(newUser as Context))
       }
     }),
 
-    removeUser: monitor(() => {
-      user = {}
+    getUser: monitor(userContextManager.getContext),
+
+    setUserProperty: monitor((key, property) => {
+      const sanitizedProperty = sanitizeUser({ [key]: property })[key]
+      userContextManager.setContextProperty(key, sanitizedProperty)
     }),
 
+    removeUserProperty: monitor(userContextManager.removeContextProperty),
+
+    /** @deprecated: renamed to clearUser */
+    removeUser: monitor(userContextManager.clearContext),
+    clearUser: monitor(userContextManager.clearContext),
+
     startView,
 
     startSessionReplayRecording: monitor(recorderApi.start),
     stopSessionReplayRecording: monitor(recorderApi.stop),
   })
   return rumPublicApi
 
-  function sanitizeUser(newUser: unknown) {
-    if (typeof newUser !== 'object' || !newUser) {
-      return
-    }
-    const result = deepClone(newUser as Context)
-    if ('id' in result) {
-      result.id = String(result.id)
+  function sanitizeUser(newUser: Context) {
+    const shallowClondeUser = assign(newUser, {})
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏typo, if you don't already use one, having some spell checker in your IDE could be handy 😉
```suggestion
    const shallowClonedUser = assign(newUser, {})
```